### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by cgrigis** [google/exposure-notifications-internals#13]
I am trying to build the project in Android Studio 4.0.1 on a Linux host, following the instructions from `BUILD.md`, but running into a few issues.
After building BoringSSL and NanoPB, and setting their locations in `build.gradle`, I got a first error, which I was able to fix with this patch (I'll make a full PR when the build succeeds):

```diff
--- a/exposurenotification/build.gradle
+++ b/exposurenotification/build.gradle
@@ -93,7 +93,7 @@ protobuf {
         artifact = 'com.google.protobuf:protoc:3.11.4'
     }
     dependencies {
-        protobuf files("${depsRoot}/nanopb-0.3.9.2-macosx-x86/generator/proto/")
+        protobuf files("${nanopbRoot}/generator/proto/")
     }
     generateProtoTasks {
         all().each { task ->
```

Now I am getting the following error:

```
/home/cgrigis/C4DT/Projects/exposure-notifications-internals/exposurenotification/src/main/cpp/CMakeLists.txt : C/C++ debug|arm64-v8a : CMake Error at /home/cgrigis/C4DT/Projects/exposure-notifications-internals/exposurenotification/src/main/cpp/CMakeLists.txt:62 (add_library):
  Cannot find source file:

    gen/exposure_key_export.pb.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx
```

It seems the `execute_process()` in `exposurenotification/src/main/cpp/CMakeLists.txt` is not triggered... any ideas?

**Comment by nic0lette** [comment]
Oops! Good catch!

As for the build failing, I never had the time to really dig into why cmake sometimes failed, but when I had a build fail, I'd go to the root of the project and run: 

```
find . -name ".cxx" -type d -exec rm -rf {} \;
```

Then I'd choose Build > Rebuild project. -- It's a pretty big hammer, but it seems to have worked for me? Let me know?

**Comment by cgrigis** [comment]
@nic0lette Thank you for your reply!

Unfortunately, the hammer did not make it work for me. :-(

I finally managed to track it down to the following changes:
* `./gen` did not exist, so I force a creation
* the protobuf executable is just `protoc` on non-WIN32 (macOS too I believe)

``` diff
--- a/exposurenotification/src/main/cpp/CMakeLists.txt
+++ b/exposurenotification/src/main/cpp/CMakeLists.txt
@@ -36,10 +36,14 @@ include_directories(matching ${BORING_SSL_ROOT}/include)
 if(WIN32)
     set(PROTOC "protoc.exe")
 else()
-    set(PROTOC "protoc.exe")
+    set(PROTOC "protoc")
 endif()
 
 # Generate the NanoPB C source and headers
+execute_process(
+        COMMAND mkdir -p ./gen
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        )
 execute_process(
         COMMAND ${NANOPB_ROOT}/generator-bin/${PROTOC} -I ${NANOPB_ROOT} -I ${NANOPB_ROOT}/generator/proto/ -I ../proto --nanopb_out=./gen ../proto/exposure_key_export.proto
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
```

I combined all my changes in #15.

**Comment by nic0lette** [comment]
Hey, I wonder if you could take a peek at the latest code pushed up that uses Docker for building BoringSSL (and includes a script for pulling down NanoPB)?

**Comment by cgrigis** [comment]
Tried it, and all works well. Thanks. :)

**Comment by cgrigis** [comment]
Fixed in da29e8f8ec2a3e970e1a435d5e40013b1a0e6172
